src/mapWithFeedback.ts

```diff
@@ -0,0 +1,147 @@
+ * @param reducer the callback function
+ * @param initialValue the initial value of the accumulator
+ * @returns An array of successively reduced values from the left side of the array.
+ * @signature
+ *    R.mapWithFeedback(items, fn, initialValue)
+ *    R.mapWithFeedback.indexed(items, fn, initialValue)
+ * @example
+ *    R.mapWithFeedback([1, 2, 3, 4, 5], (acc, x) => acc + x, 100) // => [101, 103, 106, 110, 115]
+ *    R.mapWithFeedback.indexed([1, 2, 3, 4, 5], (acc, x, i, array) => acc + x, 100) // => [101, 103, 106, 110, 115]
+ * @dataFirst
+ * @indexed
+ * @pipeable
+ * @category Array
+ */
+export function mapWithFeedback<T, Accumulator>(
+  array: ReadonlyArray<T>,
+  reducer: Reducer<T, Accumulator>,
+  initialValue: Accumulator
+): Array<Accumulator>;
+
+/**
+ * Applies a reducer function on each element of the array, accumulating the results,
+ * and returns an array of the successively reduced values.
+ * @param reducer the callback function
+ * @param initialValue the initial value to use as an accumulator value in the callback function
+ * @returns An array of successively reduced values from the left side of the array.
+ * @signature
+ *    R.mapWithFeedback(fn, initialValue)(array)
+ * @example
+ *    R.pipe([1, 2, 3, 4, 5], R.mapWithFeedback((acc, x) => acc + x, 100)) // => [101, 103, 106, 110, 115]
+ *    R.pipe([1, 2, 3, 4, 5], R.mapWithFeedback.indexed((acc, x, i, array) => acc + x, 100)) // => [101, 103, 106, 110, 115]
+ * @dataLast
+ * @indexed
+ * @pipeable
+ * @category Array
+ */
+export function mapWithFeedback<T, Accumulator>(
+  reducer: Reducer<T, Accumulator>,
+  initialValue: Accumulator
+): (items: ReadonlyArray<T>) => Array<Accumulator>;
+
+export function mapWithFeedback() {
+  return purry(
+    mapWithFeedbackImplementation(false),
+    arguments,
+    mapWithFeedback.lazy
+  );
+}
+
+const mapWithFeedbackImplementation =
+  (indexed: boolean) =>
+  <T, Accumulator>(
+    items: Array<T>,
+    reducer: ReducerIndexedOptional<T, Accumulator>,
+    initialValue: Accumulator
+  ) => {
+    const implementation = indexed
+      ? mapWithFeedback.lazyIndexed
+      : mapWithFeedback.lazy;
+
+    return _reduceLazy(items, implementation(reducer, initialValue), indexed);
+  };
+
+const lazyImplementation =
+  (indexed: boolean) =>
+  <T, Accumulator>(
+    reducer: ReducerIndexedOptional<T, Accumulator>,
+    initialValue: Accumulator
+  ) => {
+    let accumulator = initialValue;
+    const modifiedReducer: PredIndexedOptional<T, Accumulator> = (
+      currentValue,
+      index,
+      items
+    ) => {
+      accumulator = indexed
+        ? reducer(accumulator, currentValue, index, items)
+        : reducer(accumulator, currentValue);
+      return accumulator;
+    };
+
+    return (
+      value: T,
+      index?: number,
+      items?: Array<T>
+    ): LazyResult<Accumulator> => ({
+      done: false,
+      hasNext: true,
+      next: modifiedReducer(value, index, items),
+    });
```

Comment:
**question:** why introduce an additional closure here?

```suggestion
    let accumulator = initialValue;

    return (
      value: T,
      index?: number,
      items?: Array<T>
    ): LazyResult<Accumulator> => ({
      done: false,
      hasNext: true,
      next: indexed
        ? reducer(accumulator, value, index, items)
        : reducer(accumulator, value),
    });
```

Reply:
That was probably something I forgot to factor out after I made changes to other parts of the code. There is no reason for an additional closure there in this case.